ci_scripts/tag_and_release.py

```diff
@@ -1,107 +1,129 @@
-    subprocess.check_call(['git', 'add', 'docs/news/*'])
-    message = ':checkered_flag: :newspaper: releasing version %s\n[skip ci]' % version
-    subprocess.check_call(['git', 'commit', '-m', message])
-
-    if mode == release_target_map['prod']:
-        print('git - pushing %s changelog commit' % mode.name)
-        subprocess.check_call(['git', 'push', 'origin'])
-
-    # tags
-    subprocess.check_call(['git', 'tag', '-a', version, '-m', 'release %s' % version])
-    subprocess.check_call(['git', 'tag', '-f', 'latest'])
-    if mode == release_target_map['prod']:
-        print('git - pushing %s tags' % mode.name)
-        subprocess.check_call(['git', 'push', '-f', 'origin', '--tags'])
-
-    print('pypi - uploading')
-    subprocess.check_call(['python', '-m', 'twine', 'upload', mode.bundle])
-    print('pypi - uploading successful')
+    twine_repo = os.getenv(f'{ENVVAR_TWINE_REPOSITORY_URL}') or os.getenv(
+        ENVVAR_TWINE_REPOSITORY)
+    twine_username = os.getenv(ENVVAR_TWINE_USERNAME)
+
+    if not gh_token:
+        raise ValueError(
+            f'Environment variable [{ENVVAR_GIT_TOKEN}] (github token) is not set.'
+        )
+
+    if not twine_repo or not twine_username:
+        error_message = f'Environment variables [{ENVVAR_TWINE_REPOSITORY}/{ENVVAR_TWINE_REPOSITORY_URL}] (PyPI repository/URL)'
+        + f' and/or [{ENVVAR_TWINE_USERNAME}] (PyPI username) are not set.'
+        raise ValueError(error_message)
+
+
+def _release_to_pypi():
+    logger.info('Releasing to PyPI')
+    logger.info('Generating a release package')
+    with cd(PROJECT_ROOT):
+        subprocess.check_call(
+            [sys.executable, 'setup.py', 'clean', '--all',
+             'bdist_wheel',
+             '--dist-dir', 'release-dist'])
+        logger.info('Uploading to PyPI')
+        subprocess.check_call(
+            [sys.executable, '-m', 'twine', 'upload', '/release-dist/*'])
+
+
+def _determining_commit_type(is_release: bool, is_beta: bool) -> CommitType:
+    commit_type = CommitType.DEVELOPMENT
+    if is_release:
+        commit_type = CommitType.RELEASE
+    if is_beta:
+        commit_type = CommitType.BETA
+    return commit_type
+
+
+def main() -> int:
+    """Commands.
+
+    Returns:
+        success code (0) if successful; failure code otherwise.
+    """
+    parser = argparse.ArgumentParser(
+        description='Releases the project.')
+    parser.add_argument('-r', '--release',
+                        help='Performs an official release',
+                        action='store_true')
+    parser.add_argument('-b', '--beta',
+                        help='Performs a pre-release',
+                        action='store_true')
+    parser.add_argument("-v", "--verbose", action="count", default=0,
+                        help="Verbosity, by default errors are reported.")
+    args = parser.parse_args()
+    commit_type = _determining_commit_type(args.release, args.beta)
```

Comment:
Is this ever going to be called w/o `--beta` or `--release`? If not, why not have a required argument `--release-type`. It'd simplify the `_determining_commit_type`, as it'd end up being as simple mapping of release type to commit type.